misk/src/main/kotlin/misk/environment/Deployment.kt

```diff
@@ -0,0 +1,39 @@
+package misk.environment
+
+/** Deployment describes the context in which the application is running */
+data class Deployment(
+  /**
+   * The name of this deployment. This is used for debugging and should not be parsed.
+   *
+   * All pods in the same deployment will have this same name.
+   */
+  val name: String,
+
+  /**
+   * Whether the service is running in a production environment, having an SLA or handling customer data.
+   */
+  val isProduction: Boolean = false,
+
+  /**
+   * Whether the service is running in a test environment, either locally or in a CI.
+   */
+  val isTest: Boolean = false,
+
+  /**
+   * Whether the service is running on a local developer machine, including as a Docker image.
+   */
+  val isLocalDevelopment: Boolean = false
+) {
+  init {
+    if (isProduction) check(!isTest && !isLocalDevelopment)
+    if (isTest) check(!isLocalDevelopment)
+  }
+
+  /** Returns true if running in a managed cluster. Mutually exclusive with isFake. */
```

Comment:
Maybe mention staging environments?

Maybe mention that 'real' means that the code that’s running in process, and the infrastructure it’s sitting on, is also real? Real means the same components as production, whether or not there’s the same data or SLA.